lib/usagereporter/usagereporter_test.go

```diff
@@ -279,7 +279,6 @@ func TestUsageReporterDiscard(t *testing.T) {
 
 	// Wait the regular submit delay (to ensure submit finishes), as we have
 	// enough events to fill two batches.
-	fakeClock.BlockUntil(1)
```

Comment:
This test times out ~100% of the time locally for me without this, but passes 100000 times with this change.

It seems that stopped timers no longer could as a waiter/sleeper, and they did count in previous versions.

@timothyb89 or @espadolini can you take a look and let me know if I'm doing the right thing here?

Reply:
LGTM, I think the old version of clockwork didn't really like how we play with the timer quite a bit.

@timothyb89 we should simplify the logic so that we just hold a timer at a fixed interval, and then we just check if the batch is old enough to be sent, rather than relying on the timer to trigger exactly when it's old enough. wdyt?

Reply:
Yeah, that's worth a shot, anything that makes this easier to reason about is a win. The exact sequencing of clock advances/blocks/etc made sense to me once and probably never will again.